redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/composeProtocolGeneration.kt

```diff
@@ -704,14 +705,15 @@ internal fun generateComposeProtocolModifierSerialization(
         .addModifiers(INTERNAL)
         .receiver(Redwood.Modifier)
         .addParameter("json", KotlinxSerialization.Json)
-        .returns(LIST.parameterizedBy(Protocol.ModifierElement))
+        .returns(ARRAY.parameterizedBy(Protocol.ModifierElement))
         .beginControlFlow("return %M", Stdlib.buildList)
         .addStatement(
           "this@%L.forEach { element -> add(element.%M(json)) }",
           name,
           schema.modifierToProtocol,
         )
         .endControlFlow()
+        .addCode("⇥.toTypedArray()⇤")
```

Comment:
Not convinced that just adding this actually helps perf since it's still building a list first. But I think to build an array directly, `Modifier` would need a `size` and an iterator.

Reply:
We control `Modifier` so we could add behavior to expose the size if we need to.